src/renderer/components/+preferences/select-metrics-dialog.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import { observer } from "mobx-react";
+import { action, observable } from "mobx";
+import { Dialog, DialogProps } from "../dialog/dialog";
+import { Checkbox } from "../checkbox/checkbox";
+import { Wizard, WizardStep } from "../wizard/wizard";
+import { userStore } from "../../../common/user-store";
+import { Select, SelectOption } from "../select/select";
+import { Icon } from "../icon/icon";
+
+interface Props extends Partial<DialogProps> {
+}
+
+export enum ResourceType {
+  Cluster = "Cluster",
+  Node = "Node",
+  Pod = "Pod",
+  Deployment = "Deployment",
+  StatefulSet = "StatefulSet",
+  Container = "Container",
+  Ingress = "Ingress",
+  VolumeClaim = "VolumeClaim",
+  ReplicaSet = "ReplicaSet",
+  DaemonSet = "DaemonSet",
+}
+
+@observer
+export class SelectMetricsDialog extends React.Component<Props> {
+
+  @observable static isOpen = false;
+
+  static open() {
+    SelectMetricsDialog.isOpen = true;
+  }
+
+  static close() {
+    SelectMetricsDialog.isOpen = false;
+  }
+
+  close = () => {
+    SelectMetricsDialog.close();
+  };
+
+  @action
+  onChange(value: ResourceType) {
+    userStore.hiddenMetrics.has(value) ? userStore.hiddenMetrics.delete(value) : userStore.hiddenMetrics.add(value);
```

Comment:
Given that we are not using the value of this action. I don't see why we need to use the ternary.

```suggestion
    if (userStore.hiddenMetrics.has(value)) {
      userStore.hiddenMetrics.delete(value);
    } else {
      userStore.hiddenMetrics.add(value);
    }
```